File: golang/lib/api_container_context/api_container_context.go
package api_container_context

type APIContainerContext struct {
	containerID  string
	ipInsideEnclave string
	portInsideEnclave uint32
	ipOnHostMachine string
	portOnHostMachine uint32
}

func NewAPIContainerContext(containerID string, ipInsideEnclave string, portInsideEnclave uint32, ipOnHostMachine string, portOnHostMachine uint32) *APIContainerContext {
	return &APIContainerContext{containerID: containerID, ipInsideEnclave: ipInsideEnclave, portInsideEnclave: portInsideEnclave, ipOnHostMachine: ipOnHostMachine, portOnHostMachine: portOnHostMachine}
}

func (apiContext APIContainerContext) GetContainerID() string {
	return apiContext.containerID
}

func (apiContext APIContainerContext) GetIPInsideEnclave() string {
	return apiContext.ipInsideEnclave
}

func (apiContext APIContainerContext) GetPortInsideEnclave() uint32 {
	return apiContext.portInsideEnclave
}

func (apiContext APIContainerContext) GetIPOnHostMachine() string {
	return apiContext.ipOnHostMachine
}

func (apiContext APIContainerContext) GetPortOnHostMachine() uint32 {
	return apiContext.portOnHostMachine
}

File: typescript/src/lib/api_container_context/api_container_context.ts

// TODO Combine this into enclave context, now that enclave contexts can no longer be created if they don't have a valid API container!
export class ApiContainerContext {
    private readonly containerId:  string
    private readonly ipInsideEnclave: string
    private readonly portInsideEnclave: number
    private readonly ipOnHostMachine: string
    private readonly portOnHostMachine: number

    constructor(containerId: string, ipInsideEnclave: string, portInsideEnclave: number, ipOnHostMachine: string, portOnHostMachine: number) {
        this.containerId = containerId;
        this.ipInsideEnclave = ipInsideEnclave;
        this.portInsideEnclave = portInsideEnclave;
        this.ipOnHostMachine = ipOnHostMachine;
        this.portOnHostMachine = portOnHostMachine;
    }

    public getContainerId(): string {
        return this.containerId;
    }

    public getIpInsideEnclave(): string {
        return this.ipInsideEnclave;
    }

    public getPortInsideEnclave(): number {
        return this.portInsideEnclave;
    }

    public getIpOnHostMachine(): string {
        return this.ipOnHostMachine;
    }

    public getPortOnHostMachine(): number {
        return this.portOnHostMachine;
    }
}
